### PR TITLE
Implement nan-argmin/argmax

### PR DESCRIPTION
**Pull request by yurodiviy**
An attempt to cover more functions from #2077.

It seems that I need to support some form of branching (if we want to match the behaviour of NumPy in raising an error when some slice consists of NaNs), however using lax.cond feels like a bit of an overkill (and as far as my understanding goes, it would lead to multiple compilations). Any advice on what would be a good way to proceed?

### CODE REVIEW

**Comment by shoyer** [comment]
Rather than raising an error for invalid inputs, in JAX we generally try to return invalid values (e.g., NaN). Would that make sense here?

**Comment by yurodiviy** [comment]
Yep, returning NaN for all-NaN slices would make sense here and it seems to be a straightforward change. Thanks!

**Comment by shoyer** [comment]
One challenge here is that `nanargmin` returns an array with an integer dtype. If we return NaN, that gets converted into a float dtype array instead.

I'm not really sure -- there aren't any great options here -- but I think the best we can do might be to return a sentinel value like `-1`, and add a warning to both the docstring and the JAX "gotchas" page, e.g., alongside [out of bounds indexing](https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#%F0%9F%94%AA-Out-of-Bounds-Indexing).

**Comment by yurodiviy** [comment]
I didn't realize that using NaN would force everything to cast to float! Makes sense now that I'm thinking about it, since there isn't an int NaN.

I'll update the PR to return the sentinel value and add a warning about this behavior.

File: jax/numpy/lax_numpy.py [hunk 1]
```diff
@@ -2627,6 +2640,14 @@ def argmin(a, axis=None):
   return _argminmax(min, a, axis)
 
 
+@_wraps(onp.nanargmin, lax_description=_NANARG_DOC.format("min"))
+def nanargmin(a, axis=None):
+  nan_mask = isnan(a)
+  a = where(nan_mask, inf, a)
```
Comment (hawkinsp) [inline jax/numpy/lax_numpy.py]: `np.nanargmin` seems to support integer values, not just float values. Won't this coerce ints to floats? I think you need to add handling for types that aren't subtypes of `np.inexact`, e.g., by just delegating to `argmin`. Similarly for `argmax`.

**Review by hawkinsp — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: jax/numpy/lax_numpy.py (hunk 1 above)
Comment (yurodiviy) [reply jax/numpy/lax_numpy.py]: Not really sure if this would cause any bugs, but it's definitely a good idea to avoid unnecessary casts. Thanks!

**Comment by hawkinsp** [comment]
Looks good. Sorry for the slow review.

**Pull request merged**